packages/mgt/src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -695,10 +725,23 @@ export class MgtPeoplePicker extends MgtTemplatedComponent {
           }
         }
 
-        if ((this.type === PersonType.group || this.type === PersonType.any) && people.length < this.showMax) {
-          people = [];
+        if (this.type === PersonType.group && people.length < this.showMax) {
```

Comment:
IMO, we still need the check for `PersonType.any` here. The thinking is that if the type is any, and the user search above didn't come up with enough people, then do a group search as well.

Reply:
We are doing a group search in `findUsersFromGroup` already if the type is `PersonType.any`. Basically, we don't specify the `/microsoft.graph.user` or `microsoft.graph.group` endpoint, so it does a search for both. Redoing it this section will bring the groups twice, actually.

Reply:
That's kinda cool :). But that's only if `groupId` is specified. It looks like when no `groupId` is specified, we use `findUsers` - should we update that method as well to support the type?

Reply:
`findUsers` only returns users and `findGroups` only return groups. 

In our case, when no `groupId` specified and if the type is for `person` or `any`, we will be returning `/me/people` matching the `input` and if this won't be enough, we'll query the `/users`. This means only `microsoft.graph.user` values will be returned. Using the type on `findUsers` won't be useful as it can't be used otherwise.

If the type is group` or `any` and we still don't have enough users, we need to load only groups. In that case, only we use `findGroups` returning the `/groups` endpoint. So I don't think we need it here.

I'd say we're good to go here!

Reply:
Just tested this and I'm still convinced the check for `any` should be there. Here is how I tested this:

1. Created a new group ("Library Assist") through graph explorer using [this example](https://docs.microsoft.com/en-us/graph/api/group-post-groups?view=graph-rest-1.0&tabs=http#example-1-create-a-microsoft-365-group):

1. Added the following code to my page:

    ```html
     <mgt-people-picker type="group"></mgt-people-picker>
    ```

1. Searched for "Library" and **was** able to find the group

1. Change the type to `any` in step 2

1. Searched for "Library" again but this time I **was not** able to find the group

IMO, `type="any"` should be able to find both users and groups. With this code, when no `groupId` is provided, the flow is as follows:
1. the search starts with `findPeople` which does not find a user or group because either the group is not in the current signed in user personal graph
2. because there are no people found, the search continues with the `findUsers` method which only searches users and does not find any results
3. Finally, we don't have any results yet, but because the type is not set to group, we don't search the groups api and we return no results

Adding the check for `any` will ensure we search groups as well if no people are found in step 1 and 2

Reply:
Actually, this was a mistake from my side! Missed the "no group-id" scenario for the `any` part! The issue is that it might generate duplicates (groups found in the first section because we can find the users there in `findUsersFromGroup` and also again in `findGroupFromGroups`. Before returning I'm currently deduping the array.

Now the questions... Should `findUsersFromGroups` return groups when called with `any`... Right now we don't limit it to "only" users... Should we force it? The `personType` param might not be important anymore in that case... But it would mean that with "any" we would not find contacts, for instance.